OctopusDSC/DSCResources/cTentacleAgent/cTentacleAgent.psm1

```diff
@@ -461,8 +461,17 @@ function New-Tentacle {
     if ($communicationMode -eq "Listen") {
         $windowsFirewall = Get-Service -Name MpsSvc
         if ($windowsFirewall.Status -eq "Running") {
-            Write-Verbose "Open port $port on Windows Firewall"
-            Invoke-AndAssert { & netsh.exe advfirewall firewall add rule protocol=TCP dir=in localport=$port action=allow name="Octopus Tentacle: $Name" }
+			# Check to see if the firewall rule already exists
+            $rules = Invoke-AndAssert {& netsh.exe advfirewall firewall show rule name="Octopus Tentacle: $Name"}
```

Comment:
OK, we have an issue here in that if the rule does not exist, `netsh.exe` returns an exit code of 1, so `Invoke-AndAssert` throws, halting the install.

I think instead of Invoke-AndAssert here, we need a try/catch

Reply:
Ah!  Yes, I see the helper function is coded to that.  I'll see what I can do.

Reply:
I submitted another commit, hopefully this'll resolve that issue.

Reply:
:+1: lovely stuff. I'll pull that into the test VM and see how we go.